File: MVCBootstrapNavbar/Source Code/MVCBootstrapNavbar.VB/Utilities/Utilities.vb
Imports System.Runtime.CompilerServices

Namespace Utilities

    Public Module Utilities

        <Extension>
        Public Function IsActive(html As HtmlHelper, control As String, action As String) As String
            Dim routeData = html.ViewContext.RouteData

            Dim routeAction = CStr(routeData.Values("action"))
            Dim routeControl = CStr(routeData.Values("controller"))

            ' both must match
            Dim returnActive = control = routeControl AndAlso
                               action = routeAction

            Return If(returnActive, "active", "")
        End Function



    End Module
End Namespace

File: samples/navbar-active-class-mvc/MVCBootstrapNavbar.CS/Utilities/Utilities.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace MVCBootstrapNavbar.CS.Utilities {
    public static class Utilities
    {

        public static string IsActive(this HtmlHelper html, 
                                      string control,
                                      string action)
        {
            var routeData = html.ViewContext.RouteData;

            var routeAction = (string)routeData.Values["action"];
            var routeControl = (string)routeData.Values["controller"];
        
            // both must match
            var returnActive = control == routeControl &&
                               action == routeAction;

            return returnActive ? "active" : "";
        }

    }
}